src/python/pants/backend/python/providers/python_build_standalone/rules.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import collections.abc
+import functools
+import json
+import textwrap
+from typing import Iterable, TypedDict, cast
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PythonProvider
+from pants.backend.python.util_rules.pex import rules as pex_rules
+from pants.backend.python.util_rules.pex_environment import PythonExecutable
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolError,
+    ExternalToolRequest,
+)
+from pants.core.util_rules.external_tool import rules as external_tools_rules
+from pants.core.util_rules.system_binaries import CpBinary
+from pants.engine.fs import DownloadFile
+from pants.engine.internals.native_engine import FileDigest
+from pants.engine.internals.selectors import Get
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.global_options import NamedCachesDirOption
+from pants.option.option_types import StrListOption
+from pants.option.subsystem import Subsystem
+from pants.util.docutil import bin_name
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.resources import read_sibling_resource
+from pants.util.strutil import softwrap
+
+PBS_SANDBOX_NAME = ".python_build_standalone"
+PBS_NAMED_CACHE_NAME = "python_build_standalone"
+PBS_APPEND_ONLY_CACHES = FrozenDict({PBS_NAMED_CACHE_NAME: PBS_SANDBOX_NAME})
```

Comment:
There is potentially DRYing out both with the `PythonBuildStandaloneBinary` adhoc rules and the pyenv provider.  I would prefer to not block landing on either.